lib/analyse.ml

```diff
@@ -211,8 +209,11 @@ module Analysis = struct
                 get_opam ~cwd:dir path
                 >>= add_changed_pkg ~path ~name ~package ~old_content pkgs
             end
+          | "packages" :: _ ->
+            Fmt.failwith "Unexpected path %S in packages directory (expecting 'packages/name/name.l.m.n/{opam,files/*}')" path
```

Comment:
This could be changed to disallow even the `files` directory since https://github.com/ocurrent/opam-repo-ci/pull/336. We currently expect only the `opam` file and no other files.

Reply:
True, but I think the linting error you added is a nicer UI feedback than this one:

https://github.com/ocurrent/opam-repo-ci/blob/6f874c2bfd0cfc8ba1f9d59cf594c445216edaee/opam-ci-check/lib/lint_error.ml#L94-L98

Maybe we can consider a consolidation on a followup tho?